core/page_walker.c

```diff
@@ -654,22 +654,26 @@ uint32_t pw_perform_page_walk(
             pdpt_gpa = first_table;
         }
 
+        char* pdpt_page_hva;
```

Comment:
nit: There are a few cosmetic issues with this declaration:
* All local variables should be declared at the beginning of the current scope, i.e. right below line 618 (C coding style).
* `char* p` => `char *p` (C coding style).
* Although `sizeof(char)` is always 1, I'd still prefer `uint8_t`, which is explicit about the size.

Sorry I overlooked them yesterday. Let me fix them and push a new commit to your branch.

Reply:
Hmm, I'm having some trouble pushing to your branch, due to an HTTPS authentication error. Hopefully I'll find a solution soon. Meanwhile, please feel free to amend your second commit yourself:

```diff
diff --git a/core/page_walker.c b/core/page_walker.c
index 9b0f1c5..ffdea4d 100644
--- a/core/page_walker.c
+++ b/core/page_walker.c
@@ -616,8 +616,6 @@ uint32_t pw_perform_page_walk(
     first_table = pw_retrieve_table_from_cr3(cr3, is_pae, is_lme);

     if (is_pae) {
-        uint8_t *pdpt_page_hva;
-
         if (is_lme) {
             pml4t_gpa = first_table;
 #ifdef CONFIG_HAX_EPT2
@@ -656,6 +654,7 @@ uint32_t pw_perform_page_walk(
             pdpt_gpa = first_table;
         }

+        char* pdpt_page_hva;
 #ifdef CONFIG_HAX_EPT2
         pdpt_page_hva = gpa_space_map_page(&vcpu->vm->gpa_space,
                                            pdpt_gpa >> PG_ORDER_4K,
```

Reply:
Done. We'll start testing this PR now. Before I merge it, I'll merge the last two commits (`git rebase -i` with `fixup`) and force-push to your branch.